### ISSUE TITLE
ToggleSwitch binding does not work if property changed from toggled event 

### ISSUE THREAD

**Issue by KWodarczyk** [microsoft/microsoft-ui-xaml#8680]
### Describe the bug

ToggleSwitch binding does not work if property changed from toggled event.

Below i expect the toggle to switch back to it's original state but looks like the toggled event overrides the OnPropertyChanged  

```
<Page
    x:Class="ToggleSwitchBug.BlankPage1"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:local="using:ToggleSwitchBug"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    mc:Ignorable="d"
    Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">

    <StackPanel HorizontalAlignment="Center" VerticalAlignment="Center">

        <ToggleSwitch x:Name="myToggle" Toggled="myToggle_Toggled" IsOn="{Binding MyIsOnProperty}"/>

        <TextBlock Margin="20">
            <Run Text="IsOn value:"/>
            <Run Text="{Binding MyIsOnProperty}"/>
        </TextBlock>

    </StackPanel>
</Page>

```

```
public sealed partial class BlankPage1 : Page, INotifyPropertyChanged
    {
        public BlankPage1()
        {
            this.InitializeComponent();
            DataContext = this;
            MyIsOnProperty = false;
        }

        private bool _isOn;

        public bool MyIsOnProperty
        {
            get { return _isOn; }
            set
            {
                if (_isOn != value)
                {
                    _isOn = value;
                    OnPropertyChanged();
                }
            }
        }

        public event PropertyChangedEventHandler? PropertyChanged;

        /// <summary>
        /// Call events for property changed
        /// </summary>
        /// <param name="propertyName">Changed property name. Calling property name by default</param>
        protected void OnPropertyChanged([CallerMemberName] string propertyName = null)
        {
            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
        }

        private void myToggle_Toggled(object sender, RoutedEventArgs e)
        {
            myToggle.Toggled -= myToggle_Toggled;
            MyIsOnProperty = !myToggle.IsOn;
            myToggle.Toggled += myToggle_Toggled;

        }
    }
```

### Steps to reproduce the bug

copy the code and start clicking on the toggle

### Expected behavior

toggle IsOn should remain unchanged as its reset in the Toggled event via MyIsOnProperty

### Screenshots

![image](https://github.com/microsoft/microsoft-ui-xaml/assets/5243438/707f9a71-5a9e-407f-9f5a-917ad36c9db0)

### NuGet package version

WinUI 3 - Windows App SDK 1.3.2: 1.3.230602002

### Windows version

Windows 11 (22H2): Build 22621

**Comment by Jay-o-Way** [comment]
Disclaimer: i am no professional but something tells me there is something wrong with this approach. Feels like you (or your code) are either doing the same thing twice or fighting against itself. Or something like that - difficult to put in words.

Also, can you please explain **why** one would choose this construction? Why use the `-=` and `+=` lines? Why not use a single binding?

**Comment by KWodarczyk** [comment]
Sure I understand that this sound odd and it probably is but if this is bad design then the documentation should state that somewhere.

As to why to use it I will give you example.

Let's say you have a toggle switch, user toggles it but before you compete the action you want to present a confirmation popup with yes or cancel command now if user clicks cancel you want this toggle to remain as it was by reseting the property it's binding on but even if you do it will now work and the toggle will switch. Then you have to switch it back by calling .IsOn in code behind which is even worse casue you are mixing binding and direct manipulation. Oh and btw -= and += its ther cause change in the binding will trigger toggled event so we will get infinite loop. Which is another annoying thing about this Control. It should be optional.

**Comment by Jay-o-Way** [comment]
Yeah okay. The issue is that the Toggled event is called when the action is finished, and there isn't a `Toggling` event. A number of other events do have this start/end difference, like e.g. **Loading** versus **Loaded**.
Are you sure that if you set `bind ... Mode="TwoWay"` and (possibly) remove the -= and += lines, it doesn't work?

**Comment by Jay-o-Way** [comment]
Or, maybe you are _actually_ negating value twice in the code. Once **OnPropertyChanged()** and once again via **myToggle_Toggled()**.

**Comment by Jay-o-Way** [comment]
P.S. maybe you would want to use `{x:Bind}` instead.
https://learn.microsoft.com/windows/uwp/xaml-platform/x-bind-markup-extension

> {x:Bind} has a default mode of OneTime, unlike {Binding}, which has a default mode of OneWay.

**Comment by KWodarczyk** [comment]
I will tryout and see
